src/smspark/bootstrapper.py

```diff
@@ -365,26 +366,19 @@ def set_yarn_spark_resource_config(self) -> None:
         logging.info("Configuration at {} is: \n{}".format(spark_config.path, spark_config_string))
 
     def get_yarn_spark_resource_config(
-        self, instance_count: int, instance_mem_mb: int, instance_cores: int
+            self, instance_count: int, instance_mem_mb: int, instance_cores: int
     ) -> Tuple[Configuration, Configuration]:
         aws_region = os.getenv("AWS_REGION")
         executor_cores = instance_cores
         executor_count_per_instance = int(instance_cores / executor_cores)
         executor_count_total = instance_count * executor_count_per_instance
         default_parallelism = instance_count * instance_cores * 2
 
-        # Let's leave 3% of the instance memory free
-        instance_mem_mb = int(instance_mem_mb * 0.97)
-
-        driver_mem_mb = 2 * 1024
-        driver_mem_ovr_pct = 0.1
-        driver_mem_ovr_mb = int(driver_mem_mb * driver_mem_ovr_pct)
-        executor_mem_ovr_pct = 0.1
-        executor_mem_mb = int(
-            (instance_mem_mb - driver_mem_mb - driver_mem_ovr_mb)
-            / (executor_count_per_instance + executor_count_per_instance * executor_mem_ovr_pct)
-        )
-        executor_mem_ovr_mb = int(executor_mem_mb * executor_mem_ovr_pct)
+        driver_mem_mb = int(instance_mem_mb * constants.DRIVER_MEM_INSTANCE_MEM_RATIO)
+        driver_mem_overhead_mb = int(driver_mem_mb * constants.DRIVER_MEM_OVERHEAD_PCT)
+        executor_mem_mb = int(((instance_mem_mb * constants.EXECUTOR_MEM_INSTANCE_MEM_RATIO)
```

Comment:
The old code looks like it subtracts the driver_mem from instance_mem.  Do you need to do that here as well?

Reply:
I don't think we can do that, since the low initial driver memory is the major problem that we are addressing here. In terms of best practices with Spark is best to keep the driver and executors in different instances, although here we are also using executors in the driver.

Reply:
I'm not too familiar with Spark - but I'm talking about the executor_mem_mb - if driver and executors are using the same instance, don't you have to subtract driver memory + overhead from the instance memory, else you are allocating more than the instance memory.  Don't you have to guarantee `driver_mem_mb + driver_mem_overhead_mb + executor_mem_mb + executor_mem_overhead_mb <= instance_mem_mb * PCT` ?  Or do they not need to add up like that

Reply:
SageMaker Processing names host with "algo-1", "algo-2", etc. SageMaker Spark container uses client mode and we run spark-submit in "algo-1". So "algo-1" host will have both executor and driver running. Is there any test been done to verify this won't cause any regression? with this configuration executor_mem + driver_mem will exceed instance memory right?

Reply:
You could override drive mem using 

configuration=[
        {
            "Classification": "spark-defaults",
            "Properties": {
                "spark.driver.memory": "4g"
            },
        }
    ]

Reply:
I'm pretty sure it doesn't cause regression. This is an arbitrary limit that was set by default and doesn't make a lot of sense. You usually want to push results to the driver to do the final writing of results or whatever you need to do. I believe these defaults are better. I think we are the heaviest users of the spark container, I have done some tests and works as expected. I don't think is a problem that driver + executor exceeds the memory of the instance because if you anyway have too much data you will run out of memory and you might be able to use swap, overcommit some memory and when pushing to driver, the executor might start releasing memory.